com.unity.netcode.gameobjects/Components/HalfVector3.cs

```diff
@@ -0,0 +1,114 @@
+using System.Runtime.CompilerServices;
+using UnityEngine;
+
+namespace Unity.Netcode.Components
+{
+    /// <summary>
+    /// Half float precision <see cref="Vector3"/>
+    /// </summary>
+    public struct HalfVector3 : INetworkSerializable
+    {
+        /// <summary>
+        /// The half float precision value of the x-axis as a <see cref="ushort"/>
+        /// </summary>
+        public ushort X => Axis[0];
+        /// <summary>
+        /// The half float precision value of the y-axis as a <see cref="ushort"/>
+        /// </summary>
+        public ushort Y => Axis[1];
+        /// <summary>
+        /// The half float precision value of the z-axis as a <see cref="ushort"/>
+        /// </summary>
+        public ushort Z => Axis[2];
+
+        public ushort[] Axis;
+
+        private HalfVector3AxisToSynchronize m_HalfVector3AxisToSynchronize;
+
+        /// <summary>
+        /// The serialization implementation of <see cref="INetworkSerializable"/>
+        /// </summary>
+        /// <typeparam name="T"></typeparam>
+        /// <param name="serializer"></param>
+        public void NetworkSerialize<T>(BufferSerializer<T> serializer) where T : IReaderWriter
+        {
+            for (int i = 0; i < 3; i++)
+            {
+                if (m_HalfVector3AxisToSynchronize.SyncAxis[i])
+                {
+                    var axisValue = Axis[i];
+                    serializer.SerializeValue(ref axisValue);
+                    if (serializer.IsReader)
+                    {
+                        Axis[i] = axisValue;
+                    }
+                }
+            }
+        }
+
+        /// <summary>
+        /// Converts the <see cref="HalfVector3"/> to a full precision <see cref="Vector3"/>
+        /// </summary>
+        /// <param name="vector3">the <see cref="Vector3"/> to store the full precision values in</param>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public void ToVector3(ref Vector3 vector3)
+        {
+            for (int i = 0; i < 3; i++)
+            {
+                if (m_HalfVector3AxisToSynchronize.SyncAxis[i])
+                {
+                    vector3[i] = Mathf.HalfToFloat(Axis[i]);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Converts a <see cref="Vector3"/> full precision to a <see cref="HalfVector3"/>
+        /// </summary>
+        /// <param name="vector3">the <see cref="Vector3"/> to convert to a half precision <see cref="Vector3"/></param>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public void FromVector3(ref Vector3 vector3)
```

Comment:
Nitpicky, but this method name feels misleading to me. I generally expect `X.FromY()` to be a static method that returns a new `X`. Having it be an instance method that changes the instance and returns `void` surprised me.

Could I suggest `AssignFromVector3()`? Or maybe even just `Assign()`? `halfFloat.Assign(myVector3)` seems intuitive to me.

Reply:
I think I like your `halfFloat.Assign(myVector3)` better too!